features/activity/cancel_try_cancel/feature.java

```diff
@@ -0,0 +1,116 @@
+package activity.cancel_try_cancel;
+
+import io.temporal.activity.Activity;
+import io.temporal.activity.ActivityCancellationType;
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityMethod;
+import io.temporal.client.ActivityCanceledException;
+import io.temporal.client.WorkflowClient;
+import io.temporal.common.RetryOptions;
+import io.temporal.failure.ActivityFailure;
+import io.temporal.failure.ApplicationFailure;
+import io.temporal.failure.CanceledFailure;
+import io.temporal.sdkfeatures.Feature;
+import io.temporal.sdkfeatures.Run;
+import io.temporal.sdkfeatures.Runner;
+import io.temporal.sdkfeatures.SimpleWorkflow;
+import io.temporal.workflow.*;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+
+@ActivityInterface
+public interface feature extends Feature, SimpleWorkflow {
+  @SignalMethod
+  void activityResult(String result);
+
+  @ActivityMethod
+  void cancellableActivity();
+
+  class Impl implements feature {
+    private CompletablePromise<String> activityResult;
+
+    @Override
+    public void workflow() {
+      activityResult = Workflow.newPromise();
+      var activities = activities(feature.class, builder -> builder
+              .setScheduleToCloseTimeout(Duration.ofMinutes(1))
+              .setHeartbeatTimeout(Duration.ofSeconds(5))
+              // Disable retry
+              .setRetryOptions(RetryOptions.newBuilder().setMaximumAttempts(1).build())
+              .setCancellationType(ActivityCancellationType.TRY_CANCEL));
+
+      // Start cancellable activity
+      var activityPromise = new AtomicReference<Promise<Void>>();
+      var scope = Workflow.newCancellationScope(() ->
+              activityPromise.set(Async.procedure(activities::cancellableActivity))
+      );
+      scope.run();
+
+      // Sleep for short time (force task turnover)
+      Workflow.sleep(1);
+
+      // Cancel the activity and confirm it gets cancelled
+      scope.cancel();
+      try {
+        activityPromise.get().get();
+        throw ApplicationFailure.newFailure("No error", "NoError");
```

Comment:
Just a touch more clear
```suggestion
        throw ApplicationFailure.newFailure("Activity should have thrown cancellation error", "NoError");
```

Reply:
Meh, clarity on a never-should-be-thrown error in a feature is hardly worth it I think. These errors have stack traces. I guess I'll add it for y'all.

Reply:
Updated